Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -0,0 +1,835 @@
+                }
+                
+                if recurrence.daysOfTheMonth.isEmpty { 
+                    dayOfMonthAction = nil
+                } else {
+                    dayOfMonthAction = switch frequency {
+                    case .yearly, .monthly: .expand
+                    case .minutely, .hourly, .daily: .limit
+                    default: nil
+                    }
+                }
+                
+                if recurrence.weekdays.isEmpty { 
+                    weekdayAction = nil
+                } else {
+                    weekdayAction = switch frequency {
+                    case .minutely, .hourly, .daily: .limit
+                    case .weekly: .expand
+                    case .monthly: 
+                        if recurrence.daysOfTheMonth.isEmpty { 
+                            .expand
+                        } else {
+                            .limit
+                        }
+                    case .yearly: if (!recurrence.daysOfTheMonth.isEmpty || !recurrence.daysOfTheYear.isEmpty || !recurrence.weeks.isEmpty) { .limit } else { .expand }
+                    }
+                }
+                
+                if recurrence.hours.isEmpty { 
+                    hourAction = nil
+                } else {
+                    hourAction = switch frequency {
+                    case .minutely, .hourly: .limit
+                    case .yearly, .monthly, .weekly, .daily: .expand
+                    }
+                }
+                
+                if recurrence.minutes.isEmpty { 
+                    minuteAction = nil
+                } else {
+                    minuteAction = switch frequency {
+                    case .minutely: .limit
+                    case .yearly, .monthly, .weekly, .daily, .hourly: .expand
+                    }
+                }
+                
+                if recurrence.seconds.isEmpty { 
+                    secondAction = nil
+                } else {
+                    secondAction = .expand
+                }
+                
+                if let range = range {
+                    rangeLowerBoundInterval = recurrence.calendar.dateInterval(of: frequency.component, for: range.lowerBound) 
+                } else {
+                    rangeLowerBoundInterval = nil
+                }
+                
+                // Create date components that enumerate recurrences without any
+                // rules applied. Retrieve the date components of the start date
+                // but leave the field for the recurrence frequency empty, so it
+                // is the only component that increases
+#if FIXED_124640028
+                // rdar://124640028
+                // https://github.com/apple/swift-foundation/issues/483
+                // Calendar.dates(byMatching: ...) does not respect repeatedTimePolicy when .nanosecond is set #483
+                let components: Calendar.ComponentSet = switch recurrence.frequency {
```

Comment:
Does it even make sense to ever include nanoseconds in these results? It's so sparsely supported in the first place. Maybe it should really just always be set to 0.

Reply:
@hristost What do you think?